src/processor.js

```diff
@@ -241,6 +241,17 @@ function getBlockRangeMap(text, node, comments) {
 	return rangeMap;
 }
 
+const codeBlockFileNameRegex = /filename=(?<quote>["'])(?<filename>.*?)\1/u;
+
+/**
+ * Parses the file name from a block meta, if available.
+ * @param {Block} block A code block.
+ * @returns {string | null | undefined} The filename, if parsed from block meta.
+ */
+function fileNameFromMeta(block) {
+	return block.meta?.match(codeBlockFileNameRegex)?.groups.filename;
```

Comment:
I think we need some kind of validation on the filename characters to ensure it's a valid filename. Spaces, in particular may cause problems.

I'm not sure if it would make sense to throw an error or just replace any unexpected characters with a `_` (or another character).

Reply:
What if someone wants to have a space in the file name? Is there a source for what ESLint considers a valid filename?

Reply:
It's not what ESLint considers a valid filename, it's what the system considers a valid filename. Because most systems require special syntax to support spaces (surround the filename in quotes), I think it's safe to assume we should not have spaces.

Reply:
👍 Added a `\s+` replacer.